user/test/com/google/gwt/emultest/java/lang/CharacterTest.java

```diff
@@ -15,25 +15,33 @@
  */
 package com.google.gwt.emultest.java.lang;
 
+import com.google.gwt.junit.DoNotRunWith;
+import com.google.gwt.junit.Platform;
 import com.google.gwt.junit.client.GWTTestCase;
 
 import java.util.Arrays;
+import java.util.function.Function;
+import java.util.function.Predicate;
+import java.util.stream.Stream;
 
 /**
  * Tests for java.lang.Character.
  */
+@DoNotRunWith(Platform.HtmlUnitBug)
```

Comment:
Can the specific failing tests be annotated with this, each with a comment about why it fails, rather than skip existing tests that pass?

Reply:
I's pretty much all the tests that call any Character.is* method. They fail because the regexp doesn't even parse in HtmlUnit 2-4. If we want any meaningful test coverage we need https://github.com/gwtproject/gwt/pull/10115 + a custom build of HtmlUnit that includes https://github.com/mozilla/rhino/pull/1848